lib/dockerregistry/types.go

```diff
@@ -68,6 +69,14 @@ type PreCheck interface {
 	Run(edges map[PromotionEdge]interface{}) error
 }
 
+// ImageSizeCheck implements the PreCheck interface and checks against
+// images that are larger than a size threshold (controlled by the
+// max-image-size flag).
+type ImageSizeCheck struct {
+	MaxImageSize    int
+	DigestImageSize DigestImageSize
```

Comment:
We already store this information inside SyncContext... why do we need it here as well?

Reply:
because the sync context isn't accessible within the check.Run() function. would it be better to just pass in the SyncContext itself?